### PR TITLE
Allow all django services to override how often they restart.

### PR DESCRIPTION
**Pull request by feanil**
In particular update the ecommerce service to restart every 3000
requests by default.  This is because we believe there are some issues
with the segment library when run in gevent mode that cause it to take
up more and more CPU over time.

Configuration Pull Request
---

Make sure that the following steps are done before merging:

  - [ ] A DevOps team member has approved the PR.
  - [ ] Are you adding any new default values that need to be overridden when this change goes live? If so:
    - [ ] Update the appropriate internal repo (be sure to update for all our environments)
    - [ ] If you are updating a secure value rather than an internal one, file a DEVOPS ticket with details.
    - [ ] Add an entry to the CHANGELOG.
  - [ ] If you are making a complicated change, have you performed the proper testing specified on the [Ops Ansible Testing Checklist](https://openedx.atlassian.net/wiki/display/EdxOps/Ops+Ansible+Testing+Checklist)?  Adding a new variable does not require the full list (although testing on a sandbox is a great idea to ensure it links with your downstream code changes).

### CODE REVIEW

**Comment by BenjiLee** [comment]
I'm looking at NewRelic reports and it looks like we handle around 50k - 100k requests per day. Would 3000 be too frequent? Sounds like you might be looking at a more specific transaction.

**Comment by feanil** [comment]
@BenjiLee that's across all gunicorn workers, so each worker would be doing less than that.  Also about 24hrs is when we first start seeing an impact so doing it more regularly than that makes sense to me.

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,8 @@
+- Role: edx_django_service
+  - Added new overridable variable `edx_django_service_gunicorn_max_requests`
+- Role: ecommerce
+  - Set default max_requests to 6000.(eg. restart gunicorn process every 3000 requests.)
```
Comment (BenjiLee) [inline CHANGELOG.md]: Did you mean to say 6000->3000?

Comment (feanil) [reply CHANGELOG.md]: Fixed.

File: playbooks/roles/edx_django_service/defaults/main.yml [hunk 2]
```diff
@@ -60,6 +60,7 @@ edx_django_service_gunicorn_port: null
 edx_django_service_gunicorn_timeout: 300
 edx_django_service_gunicorn_workers: 2
 edx_django_service_gunicorn_worker_class: 'gevent'
+edx_django_service_gunicorn_max_requests: null
```
Comment (BenjiLee) [inline playbooks/roles/edx_django_service/defaults/main.yml:63]: What's the behaviour what this is set to null?

Comment (feanil) [reply playbooks/roles/edx_django_service/defaults/main.yml:63]: The template has a conditional which will evaluate to `False` which should prevent the max_requests config from being set in that file.  Gunicorn will fall-back to its builtin default of not rebooting the gunicorn workers.

Comment (robrap) [reply playbooks/roles/edx_django_service/defaults/main.yml:63]: @feanil: I hope [30 second graceful timeout](http://docs.gunicorn.org/en/0.17.2/configure.html#graceful-timeout) is enough. ;)  Hopefully we don't need to restart while discovery is slow or down.

**Review by robrap — COMMENTED** [review]
@feanil: Thanks for doing this!

**Comment by feanil** [comment]
@robrap @BenjiLee  thanks for the review.  Any chance of getting an approval?

**Review by robrap — APPROVED** [review]
@feanil: Thanks!  Do you want to add a note and reference to this PR to the relevant tickets, in case we ever want to see if we can remove this?

**Review by BenjiLee — APPROVED** [review]

**Comment by feanil** [comment]
@robrap I think this is good to keep around generally.  It doesn't hurt and deals with a class of issues that we probably don't want to spend too much time on.

**Pull request merged**